### ISSUE TITLE
Repository url is not valid or accessible

### ISSUE THREAD

**Issue by italolelis** [zendesk/samson#532]
Hi guys,

I've just installed samson in a fresh server, I followed everything on the README. I've created the Github application + Github personal token (with the right permissions), but everytime when I try to create a project for a private repo I got this error `Repository url is not valid or accessible`. What do I need to do to fix this?

P.S.: I tried like this and got a response, so definitely nothing wrong with they personal key.

```
curl -u italolelis:my_token https://api.github.com/repos/organization/private-repo
```

Thank you very much.

**Comment by jonmoter** [comment]
Did you check your logs?

That error gets thrown if [GitRepository#valid_url?](https://github.com/zendesk/samson/blob/master/app/models/git_repository.rb#L93) fails, and it should output the error it got to your logs.  If you could check your logs and paste what you see there, that would help us figure out what's going on here.

**Comment by italolelis** [comment]
Hey @jonmoter,
Yes I did check the valid_url, the log is this one:

```
Repository Path 'https://github.com/hellofresh/tapioca' is invalid: [nil]
```

Unfortunately I don't think is really useful... 
I did the same check that the valid_url does, which is this one...

```
git -c core.askpass=true ls-remote -h https://github.com/hellofresh/tapioca
```
And this tells me `fatal: Authentication failed`, of course I didn't pass any token with this verification, but may be that is the reason that is not working, I actually don't know, may be I'm doing something wrong....

Thanks for your help.

**Comment by henders** [comment]
if you use the 'git@github.com:hellofresh/tapioca.git' as the repo url, do you get the same error? Do you generally use SSH keys to auth? Or do you generally use user/password with the https version of the repo URL?

**Comment by italolelis** [comment]
Hi @henders ,

Generally I use SSH keys. So here is what I did to make it work:

I'm running the puma server with a specific user. The verification command that samson uses to validate the git url is executed with the same user that is running puma (I didn't knew that). So after realising this I've generated a ssh key for my user and then retried to create projects, finally it worked. I don't know if this is a good behaviour but at least now is working.

Thanks everybody for helping :+1:
